### DISCUSSION TITLE
Weird issue with indexed db and a bd player

### DISCUSSION THREAD

**Discussion by Martii** [gerbera/gerbera#3184] [General]
So some clients of mine purchased a Samsung [BD-J5100](https://www.samsung.com/ca/support/model/BD-J5100/ZC/) and [BD-J5700](https://www.samsung.com/ca/support/model/BD-J5700/ZC/) *(functionally equivalent minus wifi)* and it  has UPnP DLNA access.

While jfrog was out I decided to try my distros version of gerbera for a while:

``` console
$ lsb_release -d
Description:    Ubuntu 22.04.4 LTS

$ apt-cache policy gerbera
gerbera:
  Installed: 1.1.0+dfsg-3ubuntu4
  Candidate: 1.1.0+dfsg-3ubuntu4
  Version table:
 *** 1.1.0+dfsg-3ubuntu4 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
        100 /var/lib/dpkg/status
```

Things were buggy but the indexed db was available for Kodi and the BD player. It gave me a chance to see that AVC/AC3 and AVC/AAC were supported in .mp4 and .mkv containers.

However once the new repo was up I immediately wanted to curtail the bugs from the older gerbera so I purged the distro version and set up the ppa and installed it:

``` console
$ apt-cache policy gerbera 
gerbera:
  Installed: 2.2.0-jammy1
  Candidate: 2.2.0-jammy1
  Version table:
 *** 2.2.0-jammy1 500
        500 https://pkg.gerbera.io/debian jammy/main amd64 Packages
        100 /var/lib/dpkg/status
     1.1.0+dfsg-3ubuntu4 500
        500 http://us.archive.ubuntu.com/ubuntu jammy/universe amd64 Packages
```

Kodi still works with the newly created indexed db however all listings of content ceased on the BD player.

I'm not even sure where to start from the out of the box settings in config.xml...

I would appreciate some guidance on what changed between 1.1.0 and 2.2.0 to prevent the BD players from even listing the items. I've gone ahead and again put 1.1.0 on another test machine for comparison but have failed to find the difference. The test machine is available to the BD players and finds the indexed list just fine however the other machine with the newer gerbera still does not.

Thank you in advance for any pointers.

**Comment by KarlStraussberger** [comment]
That's like upgrading from WindowsXP to Windows 11.

You should uninstall completely and clean up: https://docs.gerbera.io/en/latest/troubleshooting.html

The config may contain loads of legacy stuff, recreate with `--create-config` like a fresh instalation and copy the entries you still find in the documentation https://docs.gerbera.io/en/latest/config-overview.html

Then you can start to build the database again.

Mediatomb and early gerbera had their private branch of libpupnp which supported multiple network interfaces. That feature can only be used if your built is based on libnpupnp.

**Comment by Martii** [comment]
@KarlStraussberger 

> I immediately wanted to curtail the bugs from the older gerbera so I purged the distro version

is the same as...

> You should uninstall completely and clean up

While I appreciate the response please pay attention to wording. 

I also on a third machine, which has never seen Gerbera, installed directly from the PPA and have the same issue of it still not listing.

> The config may contain loads of legacy stuff,...

... again not an issue. The /etc/gerbera/config.xml and the db pointed to it in it was obliterated *(purged is the normal word)*.


> Then you can start to build the database again.

See previous two repsonses. Non-issue.

It does see the server, just doesn't want to list any of my AVC, HEVC, AV1 test files like 1.x does.

> That's like upgrading from WindowsXP to Windows 11.

I have Windows XP on a target platform as well. :smile_cat: 


So the original issue still remains despite a valiant effort to point the issue at user error. This is definitely an out of the box issue in Gerbera with this hardware.  So I need to know beyond what has previously been done.

Thanks.

**Comment by Martii** [comment]
I do notice a Gerbera issue with the debug log though. There are a ton of these html entity uses:

``` console
&lt;item id="46446" parentID="24475" restricted="1"&gt;
&lt;dc:title&gt;S6E22&lt;/dc:title&gt;
&lt;upnp:class&gt;object.item.videoItem&lt;/upnp:class&gt;
&lt;dc:created&gt;2008-11-19&lt;/dc:created&gt;
&lt;dc:date&gt;2001-01-01&lt;/dc:date&gt;
&lt;dc:description&gt;Test clip&lt;/dc:description&gt;
&lt;upnp:artist&gt;Me, you, everyone&lt;/upnp:artist&gt;
&lt;upnp:artist role="AlbumArtist"&gt;Richard Simmons&lt;/upnp:artist&gt;
&lt;upnp:date&gt;2001-01-01&lt;/upnp:date&gt;
&lt;upnp:genre&gt;Test&lt;/upnp:genre&gt;
&lt;upnp:albumArtURI xmlns:dlna="urn:schemas-dlna-org:metadata-1-0" dlna:profileID="JPEG_TN"&gt;http://192.168.10.92:49152/media/object_id/46446/res_id/1/ext/file.jpg&lt;/upnp:albumArtURI&gt;
&lt;upnp:albumArtURI xmlns:dlna="urn:schemas-dlna-org:metadata-1-0" dlna:profileID="JPEG_TN"&gt;http://192.168.10.92:49152/media/object_id/46446/res_id/2/ext/file.jpg&lt;/upnp:albumArtURI&gt;
&lt;res size="499295454" duration="0:44:26.373" bitrate="187256" sampleFrequency="48000" nrAudioChannels="6" resolution="720x478" protocolInfo="http-get:*:video/mp4:DLNA.ORG_PN=AVC_MP4_EU;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=01700000000000000000000000000000" bitsPerSample="16" sec:acodec="ac3" sec:vcodec="hevc"&gt;http://192.168.10.92:49152/media/object_id/46446/res_id/0/group/default/ext/file.mp4&lt;/res&gt;
&lt;res protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00800000000000000000000000000000"&gt;http://192.168.10.92:49152/media/object_id/46446/res_id/1/ext/file.jpg&lt;/res&gt;
&lt;res resolution="160x106" protocolInfo="http-get:*:image/jpeg:DLNA.ORG_PN=JPEG_TN;DLNA.ORG_OP=01;DLNA.ORG_CI=0;DLNA.ORG_FLAGS=00800000000000000000000000000000"&gt;http://192.168.10.92:49152/media/object_id/46446/res_id/2/ext/file.jpg&lt;/res&gt;
&lt;/item&gt;
```

In some areas of the log it is actually dereferenced i.e. `<` and `>`. This could present an issue with hardware if it's not producing proper XML/HTML?

So it's been logging that the listing is being done however the final output is an empty list screen from the BD players... which is why I'm here with this discussion/issue.

---

``` console
...trace: [clients.cc:247] getInfo(): client info: 192.168.10.90:60738 '[BD]J5100 DLNADOC/1.50' -> 'Unknown' as None with None
```

... this is curious too.

**Comment by KarlStraussberger** [comment]
Sorry for aswering with basic questions, but more often than not people write they did something which was simple not the real thing.

Regarding the HTML/XML: That is the UPnP protocol.

In case the client is not known to us, we do not activate any device specific extensions. You may add some Samsung client extensions for your BD in `config.xml`

```
<clients enabled="yes">
	<client ip="192.168.10.90" flags="SAMSUNG|SAMSUNG_FEATURES|SAMSUNG_BOOKMARK_SEC" allowed="yes"/>
</clients>
```

We are looking for the browse-request of the client and the

For better reading of debugging messages you can restrict them by using `--debug-mode 'request|cds'` instead of `--debug`.

**Comment by Martii** [comment]
> In case the client is not known to us, we do not activate any device specific extensions. You may add some Samsung client extensions for your BD in `config.xml`

I will most definitely give that a try. Hope it works and thank you for pointing me in a direction I wasn't familiar with. I was pondering adding a feature request to the `clients` tag but seems it's already there. :smile: .

> We are looking for the browse-request of the client and the

Is this a Rocky Horror Picture Show reference on Antici.............pation? i.e. "and the" what? :smile_cat: 

> For better reading of debugging messages you can restrict them by using `--debug-mode 'request|cds'` **instead** of `--debug`.

Will definitely try trimming the debug log with the values you mentioned, with casings stated at https://docs.gerbera.io/en/latest/troubleshooting.html#debug-mode . I just `tail`ed initially but then had to go in with a copy and `less` it because there was definitely a lot of information rolling through. 

Thanks again and will report back when it gets retested. I've been with Gerbera for as long as I can remember... missed the initial Mediatomb days though. Fresh and manually cleaned installs are things I've been doing since Windows 2.0 *(Windows 1.0 wasn't worth using but went through it too)*.

Misc note: Apologies if my avatar was missing when you first encountered this discussion. Logged in yesterday and it was missing... so reinstated it. Weird GH issue. :face_with_diagonal_mouth:

**Comment by Martii** [comment]
@KarlStraussberger

So here is some detailed information that I finally got around to doing. Real life has a tendency to throw wrenches into the workings sometimes. Apologies for this later than sooner follow-up. :smile_cat:

I've included a lot of data here on steps-to-reproduce... hope it helps *(don't forget to expand what you want to examine with the arrows/mouse)*.

## Details

<details>

<summary>

``` console

$ # Some modifications are done to transcoding as I moved ahead with further testing beyond this issue with the found .mkv files

$ cat /etc/gerbera/config.xml
```

</summary>

<p>

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<config version="2" xmlns="http://mediatomb.cc/config/2" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://mediatomb.cc/config/2 http://mediatomb.cc/config/2.xsd">
  <!--
     See https://gerbera.io or read the docs for more
     information on creating and using config.xml configuration files.
    -->
  <server>
    <ui enabled="yes" show-tooltips="yes">
      <accounts enabled="yes" session-timeout="30">
        <account user="gerbera" password="gerbera" />
      </accounts>
    </ui>
    <name>2.2.0</name>
    <udn>uuid:975ce3d9-997f-42c5-ae6a-fe95ac5cb937</udn>
    <home>/var/lib/gerbera/</home>
    <webroot>/usr/share/gerbera/web</webroot>
    <!--
        How frequently (in seconds) to send ssdp:alive advertisements.
        Minimum alive value accepted is: 62

        The advertisement will be sent every (A/2)-30 seconds,
        and will have a cache-control max-age of A where A is
        the value configured here. Ex: A value of 62 will result
        in an SSDP advertisement being sent every second.
    -->
    <alive>180</alive>
    <storage>
      <sqlite3 enabled="yes">
        <database-file>gerbera.db</database-file>
      </sqlite3>
      <mysql enabled="no">
        <host>localhost</host>
        <username>gerbera</username>
        <database>gerbera</database>
      </mysql>
    </storage>
    <containers enabled="yes">
      <container location="/LastAdded" title="Recently Added" sort="-last_updated">
        <filter>upnp:class derivedfrom "object.item" and last_updated &gt; "@last7"</filter>
      </container>
      <container location="/LastModified" title="Recently Modified" sort="-last_modified">
        <filter>upnp:class derivedfrom "object.item" and last_modified &gt; "@last7"</filter>
      </container>
    </containers>
    <extended-runtime-options>
      <ffmpegthumbnailer enabled="yes">
        <thumbnail-size>160</thumbnail-size>
        <seek-percentage>5</seek-percentage>
        <filmstrip-overlay>no</filmstrip-overlay>
        <image-quality>8</image-quality>
      </ffmpegthumbnailer>
      <mark-played-items enabled="no" suppress-cds-updates="yes">
        <string mode="prepend">*</string>
        <mark>
          <content>video</content>
        </mark>
      </mark-played-items>
    </extended-runtime-options>
  </server>
  <import hidden-files="no">
    <magic-file></magic-file>
    <scripting script-charset="UTF-8">
      <script-folder>
        <common>/usr/share/gerbera/js</common>
      </script-folder>
      <import-function>
        <playlist>importPlaylist</playlist>
        <meta-file>importMetadata</meta-file>
        <audio-file>importAudio</audio-file>
        <video-file>importVideo</video-file>
        <image-file>importImage</image-file>
        <trailer>importTrailer</trailer>
      </import-function>
      <virtual-layout type="builtin">
        <boxlayout>
          <box key="Audio/allAlbums" title="Albums" class="object.container" />
          <box key="Audio/allArtists" title="Artists" class="object.container" />
          <box key="Audio/allAudio" title="All Audio" class="object.container" />
          <box key="Audio/allComposers" title="Composers" class="object.container" />
          <box key="Audio/allDirectories" title="Directories" class="object.container" />
          <box key="Audio/allGenres" title="Genres" class="object.container" />
          <box key="Audio/allSongs" title="All Songs" class="object.container" />
          <box key="Audio/allTracks" title="All - full name" class="object.container" />
          <box key="Audio/allYears" title="Year" class="object.container" />
          <box key="Audio/audioRoot" title="Audio" class="object.container" />
          <box key="Audio/artistChronology" title="Album Chronology" class="object.container" />
          <box key="AudioInitial/abc" title="ABC" class="object.container" />
          <box key="AudioInitial/allArtistTracks" title="000 All" class="object.container" />
          <box key="AudioInitial/allBooks" title="Books" class="object.container" />
          <box key="AudioInitial/audioBookRoot" title="AudioBooks" class="object.container" />
          <box key="AudioStructured/allAlbums" title="-Album-" class="object.container" size="6" />
          <box key="AudioStructured/allArtistTracks" title="all" class="object.container" />
          <box key="AudioStructured/allArtists" title="-Artist-" class="object.container" size="9" />
          <box key="AudioStructured/allGenres" title="-Genre-" class="object.container" size="6" />
          <box key="AudioStructured/allTracks" title="-Track-" class="object.container" size="6" />
          <box key="AudioStructured/allYears" title="-Year-" class="object.container" />
          <box key="Video/allDates" title="Date" class="object.container" />
          <box key="Video/allDirectories" title="Directories" class="object.container" />
          <box key="Video/allVideo" title="All Video" class="object.container" />
          <box key="Video/allYears" title="Year" class="object.container" />
          <box key="Video/unknown" title="Unknown" class="object.container" />
          <box key="Video/videoRoot" title="Video" class="object.container" />
          <box key="Image/allDates" title="Date" class="object.container" />
          <box key="Image/allDirectories" title="Directories" class="object.container" />
          <box key="Image/allImages" title="All Photos" class="object.container" />
          <box key="Image/allYears" title="Year" class="object.container" />
          <box key="Image/imageRoot" title="Photos" class="object.container" />
          <box key="Image/unknown" title="Unknown" class="object.container" />
          <box key="Trailer/trailerRoot" title="Online Services" class="object.container" />
          <box key="Trailer/appleTrailers" title="Apple Trailers" class="object.container" />
          <box key="Trailer/allTrailers" title="All Trailers" class="object.container" />
          <box key="Trailer/allGenres" title="Genres" class="object.container" />
          <box key="Trailer/relDate" title="Release Date" class="object.container" />
          <box key="Trailer/postDate" title="Post Date" class="object.container" />
          <box key="Trailer/unknown" title="Unknown" class="object.container" />
          <box key="Playlist/playlistRoot" title="Playlists" class="object.container" />
          <box key="Playlist/allPlaylists" title="All Playlists" class="object.container" />
          <box key="Playlist/allDirectories" title="Directories" class="object.container" />
        </boxlayout>
      </virtual-layout>
    </scripting>
    <mappings>
      <extension-mimetype ignore-unknown="no">
        <map from="asf" to="video/x-ms-asf" />
        <map from="asx" to="video/x-ms-asx" />
        <map from="dff" to="audio/x-dff" />
        <map from="dsd" to="audio/x-dsd" />
        <map from="dsf" to="audio/x-dsf" />
        <map from="flv" to="video/x-flv" />
        <map from="m2ts" to="video/mp2t" />
        <map from="m3u" to="audio/x-mpegurl" />
        <map from="m3u8" to="audio/x-mpegurl" />
        <map from="m4a" to="audio/mp4" />
        <map from="mka" to="audio/x-matroska" />
        <map from="mkv" to="video/x-matroska" />
        <map from="mp3" to="audio/mpeg" />
        <map from="mts" to="video/mp2t" />
        <map from="oga" to="audio/ogg" />
        <map from="ogg" to="audio/ogg" />
        <map from="ogm" to="video/ogg" />
        <map from="ogv" to="video/ogg" />
        <map from="ogx" to="application/ogg" />
        <map from="pls" to="audio/x-scpls" />
        <map from="ts" to="video/mp2t" />
        <map from="tsa" to="audio/mp2t" />
        <map from="tsv" to="video/mp2t" />
        <map from="wax" to="audio/x-ms-wax" />
        <map from="wm" to="video/x-ms-wm" />
        <map from="wma" to="audio/x-ms-wma" />
        <map from="wmv" to="video/x-ms-wmv" />
        <map from="wmx" to="video/x-ms-wmx" />
        <map from="wv" to="audio/x-wavpack" />
        <map from="wvx" to="video/x-ms-wvx" />
        <!-- Uncomment the line below for PS3 divx support -->
        <!-- <map from="avi" to="video/divx" /> -->
        <!-- Uncomment the line below for D-Link DSM / ZyXEL DMA-1000 -->
        <!-- <map from="avi" to="video/avi" /> -->
      </extension-mimetype>
      <mimetype-upnpclass>
        <map from="application/ogg" to="object.item.audioItem.musicTrack" />
        <map from="audio/*" to="object.item.audioItem.musicTrack" />
        <map from="image/*" to="object.item.imageItem" />
        <map from="video/*" to="object.item.videoItem" />
      </mimetype-upnpclass>
      <mimetype-contenttype>
        <treat mimetype="application/ogg" as="ogg" />
        <treat mimetype="audio/L16" as="pcm" />
        <treat mimetype="audio/flac" as="flac" />
        <treat mimetype="audio/mp4" as="mp4" />
        <treat mimetype="audio/mpeg" as="mp3" />
        <treat mimetype="audio/ogg" as="ogg" />
        <treat mimetype="audio/x-dsd" as="dsd" />
        <treat mimetype="audio/x-flac" as="flac" />
        <treat mimetype="audio/x-matroska" as="mka" />
        <treat mimetype="audio/x-mpegurl" as="playlist" />
        <treat mimetype="audio/x-ms-wma" as="wma" />
        <treat mimetype="audio/x-scpls" as="playlist" />
        <treat mimetype="audio/x-wav" as="pcm" />
        <treat mimetype="audio/x-wavpack" as="wv" />
        <treat mimetype="image/jpeg" as="jpg" />
        <treat mimetype="image/png" as="png" />
        <treat mimetype="video/mkv" as="mkv" />
        <treat mimetype="video/mp4" as="mp4" />
        <treat mimetype="video/mpeg" as="mpeg" />
        <treat mimetype="video/x-matroska" as="mkv" />
        <treat mimetype="video/x-mkv" as="mkv" />
        <treat mimetype="video/x-ms-asf" as="asf" />
        <treat mimetype="video/x-ms-asx" as="playlist" />
        <treat mimetype="video/x-msvideo" as="avi" />
      </mimetype-contenttype>
      <mimetype-dlnatransfermode>
        <map from="application/ogg" to="Streaming" />
        <map from="application/x-srt" to="Background" />
        <map from="audio/*" to="Streaming" />
        <map from="image/*" to="Interactive" />
        <map from="srt" to="Background" />
        <map from="text/*" to="Background" />
        <map from="video/*" to="Streaming" />
      </mimetype-dlnatransfermode>
      <contenttype-dlnaprofile>
        <map from="asf" to="VC_ASF_AP_L2_WMA" />
        <map from="avi" to="AVI" />
        <map from="dsd" to="DSF" />
        <map from="flac" to="FLAC" />
        <map from="jpg" to="JPEG_LRG" />
        <map from="mka" to="MKV" />
        <map from="mkv" to="MKV" />
        <map from="mp3" to="MP3" />
        <map from="mp4" to="AVC_MP4_EU" />
        <map from="mpeg" to="MPEG_PS_PAL" />
        <map from="ogg" to="OGG" />
        <map from="pcm" to="LPCM" />
        <map from="png" to="PNG_LRG" />
        <map from="wma" to="WMAFULL" />
      </contenttype-dlnaprofile>
    </mappings>
    <online-content>
      <AppleTrailers enabled="no" refresh="43200" update-at-start="no" resolution="640" />
    </online-content>
  </import>
  <transcoding enabled="no">
    <mimetype-profile-mappings>
      <transcode mimetype="video/x-matroska" client-flags="TRANSCODING1" using="ffmpeg" />
    </mimetype-profile-mappings>
    <profiles>
      <profile name="ffmpeg" enabled="no" type="external">
        <mimetype>video/x-matroska</mimetype>
        <accept-url>no</accept-url>
        <first-resource>yes</first-resource>
        <hide-original-resource>yes</hide-original-resource>
        <agent command="ffmpeg" arguments="-y -i %in -c:v libx264 -preset veryfast -c:a copy -f matroska %out"/>
        <buffer size="4194304" chunk-size="524288" fill-size="1048576"/>
      </profile>
    </profiles>
  </transcoding>
  <clients enabled='yes'>
    <client ip="192.168.0.80" flags="SAMSUNG|SAMSUNG_FEATURES|SAMSUNG_BOOKMARK_SEC|TRANSCODING1" allowed="yes"/>
  </clients>
</config>
```

</p>
</details>

``` console
$ sudo systemctl stop gerbera
```

<details>
<summary>

``` console
$ sudo nano /usr/lib/systemd/system/gerbera.service
```

</summary>

<p>

``` ini
[Unit]
Description=Gerbera Media Server with MySQL
After=mysql.target network-online.target
Wants=network-online.target

[Service]
Type=simple
User=gerbera
Group=gerbera
ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml --syslog
#ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml --debug --logfile /tmp/gerbera.2.2.0.full.log
#ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml --debug-mode 'request|cds' --logfile /tmp/gerbera.2.2.0.filtered.log
Restart=on-failure
RestartSec=5

[Install]
WantedBy=multi-user.target
```

</p>
</details>

``` console
$ sudo systemctl daemon-reload
```

``` console
$ sudo systemctl start gerbera
```

### Log

* [/tmp/gerbera.2.2.0.full.log](https://github.com/user-attachments/files/16885602/gerbera.2.2.0.full.log)
* [/tmp/gerbera.2.2.0.filtered.log](https://github.com/user-attachments/files/16885608/gerbera.2.2.0.filtered.log)

## Server

<a href="https://github.com/user-attachments/assets/80c232e7-53b3-4b62-ad58-227f62367ce8"><img src="https://github.com/user-attachments/assets/80c232e7-53b3-4b62-ad58-227f62367ce8" width="576" height="272" alt="00 Server DB" title="Click to enlarge... 00 Server DB"></a>


## Client

* <a href="https://github.com/user-attachments/assets/2a1234af-583a-4962-ba53-b027c6464acf"><img src="https://github.com/user-attachments/assets/2a1234af-583a-4962-ba53-b027c6464acf" width="327" height="189" alt="01 Settings-Support" title="Click to enlarge... 01 Settings-Support"></a>
  * The up-to-date firmware revision.

* <a href="https://github.com/user-attachments/assets/560dafa7-7375-4492-b839-125ecf538455"><img src="https://github.com/user-attachments/assets/560dafa7-7375-4492-b839-125ecf538455" width="327" height="189" alt="02 Change Device-1.1.0" title="Click to enlarge... 02 Change Device-1.1.0"></a>
  * Selecting Gerbera 1.1.0 server.

* <a href="https://github.com/user-attachments/assets/fc7a2f1b-afe4-4d03-87d1-bc34e63f62f5"><img src="https://github.com/user-attachments/assets/fc7a2f1b-afe4-4d03-87d1-bc34e63f62f5" width="327" height="189" alt="03a Videos-Video Page 1 of 2" title="Click to enlarge... 03a Videos-Video Page 1 of 2"></a> <a href="https://github.com/user-attachments/assets/052aebfa-28c5-47ff-98e0-912668ee0cb1"><img src="https://github.com/user-attachments/assets/052aebfa-28c5-47ff-98e0-912668ee0cb1" width="327" height="189" alt="03b Videos-Video Page 2 of 2" title="Click to enlarge... 03b Videos-Video Page 2 of 2"></a>
  * All Video listing on Gerbera 1.1.0 server (Pages 1 and 2).

* <a href="https://github.com/user-attachments/assets/d99016e9-b736-4c00-9cba-48ee37c627cf"><img src="https://github.com/user-attachments/assets/d99016e9-b736-4c00-9cba-48ee37c627cf" width="327" height="189" alt="04 Change Device From 1.1.0 to 2.2.0" title="Click to enlarge... 04 Change Device From 1.1.0 to 2.2.0"></a>
  * Switching to Gerbera 2.2.0 server.

* <a href="https://github.com/user-attachments/assets/53adf206-cd17-432e-80e9-f1f572dd2d03"><img src="https://github.com/user-attachments/assets/53adf206-cd17-432e-80e9-f1f572dd2d03" width="327" height="189" alt="05 Videos-Video Page 1 of 1" title="Click to enlarge... 05 Videos-Video Page 1 of 1"></a>
  * All Video listing on Gerbera 2.2.0 server. Unfortunately missing .mp4's.


## Misc notes


### Features enabled

``` xml
<clients enabled="yes">
	<client ip="192.168.0.80" flags="SAMSUNG|SAMSUNG_FEATURES|SAMSUNG_BOOKMARK_SEC" allowed="yes"/>
</clients>
```

... did thankfully help with squashing the related messages from old debug log of this example:

``` log
trace: [quirks.cc:124] getSamsungFeatureRoot(): getSamsungFeatureRoot called, but it is not enabled for this client
```

... however it has no effect on the missing .mp4 files.

### Filtering enabled

`--debug-mode 'request|cds'`

... did trim the logs but did too much as all the found .mp4 files weren't logged.


## Summary

1. **I would like to know please if there is a way to bring back the .mp4 files with the out of the box Gerbera distribution, perhaps with a force mp4 listing flag in the `clients` section?**  I am also not here to debate how better one container format is compared to the other and with many of my clients, who have thousands of their creator content already in one container format and backed up several times, conversion is not an option.

2. Originally I had a test file misnamed which threw off some of my earlier testing when I created this issue.  The container was .mp4 but the file extension was .mkv. Gerbera 2.2.0 listed it and streamed it successfully to the device. I also neglected to have any .mkv on the original 2.2.0 production server but was fixed on these development servers. Also not a viable option to improperly rename thousands of creator content files.

3. Minorly I would also be interested in knowing if I can get the file extensions back that were there in 1.1.0 but not in 2.2.0. I had to prepend the uppercase extension lettering to the filenames to see what was happening.

Thanks for the relook and helping me understand the issue better.  If you need anything else please don't hesitate to ask.

**Reply by KarlStraussberger** [reply]
There might be many causes for this behaviour.

At first glance, I would try to tweak the dlna-profile in `<contenttype-dlnaprofile>` because that's how these devices determine whether they can play a file or not. 

```xml
<contenttype-dlnaprofile>
...
    <map from="mp4" to="AVC_MP4_EU"/>
    <map from="mp4" videoCodec="h264" audioCodec="aac" to="AVC_MP4_MP_HD_720p_AAC"/>
...
</contenttype-dlnaprofile>
```

I found some more values:

- HEVC_MP4
- AVC_MP4_MP_SD_AAC_MULT5
- AVC_MP4_MP_SD_MPEG1_L3
- AVC_MP4_MP_SD_AC3
- AVC_MP4_MP_HD_720p_AAC
- AVC_MP4_MP_HD_1080i_AAC
- AVC_MP4_HP_HD_AAC
- AVC_MP4_LPCM
- AVC_MP4_MP_SD_DTS
- AVC_MP4_MP_SD_DTSHD
- AVC_MP4_HP_HD_DTS
- AVC_MP4_HP_HD_DTSHD
- AVC_MP4_HP_HD_MLP

**Comment by Martii** [comment]
Thanks. After trying all of them, and improvising a bit beyond the example you gave,  this:

``` console
$ diff -c3 a/config.xml b/config.xml
```

▾

``` diff
*** a/config.xml
--- b/config.xml
***************
*** 207,212 ****
--- 207,218 ----
          <map from="mkv" to="MKV" />
          <map from="mp3" to="MP3" />
          <map from="mp4" to="AVC_MP4_EU" />
+         <map from="mp4" videoCodec="h264" audioCodec="aac" to="AVC_MP4_HP_HD_AAC"/>
+         <map from="mp4" videoCodec="av1" audioCodec="aac" to="AVC_MP4_HP_HD_AAC"/>
+         <map from="mp4" videoCodec="h264" audioCodec="ac3" to="AVC_MP4_HP_HD_AAC"/>
+         <map from="mp4" videoCodec="hevc" audioCodec="ac3" to="AVC_MP4_HP_HD_AAC"/>
+         <map from="mp4" videoCodec="mpeg4" audioCodec="ac3" to="AVC_MP4_HP_HD_AAC"/>
+         <map from="mp4" videoCodec="mpeg2video" audioCodec="ac3" to="AVC_MP4_HP_HD_AAC"/>
          <map from="mpeg" to="MPEG_PS_PAL" />
          <map from="ogg" to="OGG" />
          <map from="pcm" to="LPCM" />
```


... yields:

* <a href="https://github.com/user-attachments/assets/7756b530-a24e-41eb-8401-a78f46e94dc2"><img src="https://github.com/user-attachments/assets/7756b530-a24e-41eb-8401-a78f46e94dc2" width="327" height="189" alt="0A1 Page 1 of 2" title="Click to enlarge... 0A1 Page 1 of 2"></a> <a href="https://github.com/user-attachments/assets/fcd10eb8-bf80-4f1e-85d2-427f50aa3d3c"><img src="https://github.com/user-attachments/assets/fcd10eb8-bf80-4f1e-85d2-427f50aa3d3c" width="327" height="189" alt="0A2 Page 2 of 2" title="Click to enlarge... 0A2 Page 2 of 2"></a>


Awesome! Many Thanks to you for the pointer. Glad it was your intuition and not the scary many other possibilities that were alluded to. :smile_cat:


So a few requestioning on this workaround for this player if you could please:

1. What is "HP" in this context? AVC MP4 and AAC I can consider codec names "found" so it sends that in the list and the player then decodes that. HD and SD are probably High Definition and Standard Definition. However I have no idea what "HP" is. Do you know? For that fact why is default "EU" in there too? European Union? I am familiar with changing MIME types *(or Content-Type as it is sometimes called)* so this is interesting to know the UPnP DLNA spec can be this varied.

2. I don't see any other properties for [`map`](https://docs.gerbera.io/en/latest/config-import.html#id9) under [`contenttype-dlnaprofile`](https://docs.gerbera.io/en/latest/config-import.html#contenttype-dlnaprofile) other than `from` and `to`. This remap workaround is proably going to bite me, or someone else, at some point with a different player device. Luckily Kodi doesn't seem to care atm from what systems I've retested so far. I do think it would be highly recommended to have some way to link these modifications into/from a client specific profile so it never interferes with another device... Don't you? i.e. `client-flags` or equivalent.

3. Is there a global wildcard for these properties of `videoCodec` or `audioCodec`? i.e. `*`. I can potentially see this remap getting rather long depending on the clients used codecs. *(think `ffmpeg -codecs`)*

4. I can live without the filename extensions in the client list but it would be nice to have a toggle for those who can't. Any chance of that?


Many thanks again for the assistance and helpful answers.

**Reply by KarlStraussberger** [reply]
1. Sorry that's what someone else invented, never came across an explanation
2. In fact you can use every resource attribute here. It's very dynamic. Already started working on adding it to client configuration like mime-type and headers
3. If you are mapping to the same profile, you can skip `audioCodec`, which is the `*`
4. It's not the extension, it's the result of the mimetype-contenttype mapping. So it goes through the following steps:
  - extension-mimetype +  mime detection (if active): We're linux, extension are just part of the filename
  - mimetype-contenttype: Different mimetypes may represent the same content
  - contenttype-dlnaprofile: Make all this understandable for the player

**Reply by whyman** [reply]
1. I'm pretty sure HP is High Profile in the AVC context. But it's defined in the DLNA specs which are not publicly published, I think I have the PDFs somewhere

**Reply by Martii** [reply]
> 2. In fact you can use every resource attribute here. It's very dynamic. Already started working on adding it to client configuration like mime-type and headers

I look forward to that addition coming to Gerbera. Thank you and I do appreciate this projects work and support.

> 3. If you are mapping to the same profile, you can skip `audioCodec`, which is the `*`

Works like a charm. Thank you again.

> 4. It's not the extension, it's the result of the mimetype-contenttype mapping. So it goes through the following steps:
>  
> 
>     * ... We're linux, extension are just part of the filename

Yes but then Gerbera and the support libraries is stripping the last 4 characters of the filename then with that same logic. :smiley_cat: That changes the filename and doesn't properly represent the filename. Extensions have always been "hints" even going back to DOS/CPM/TRSDOS days. MIME types are newer comparatively but data and binaries have always had embedded signatures that can be aliased to a MIME type and a now logical extension.

> 1. I'm pretty sure HP is High Profile in the AVC context. But it's defined in the DLNA specs which are not publicly published, ...

I think you are right. With this tidbit it helped me web search and it seems to be discussed fully at https://www.wikipedia.org/wiki/Advanced_Video_Coding#Profiles :
> ...
> __Main Profile (MP, 77)__
>   * This profile is used for standard-definition digital TV broadcasts that use the MPEG-4 format as defined in the DVB standard.[[51]](https://www.wikipedia.org/wiki/Advanced_Video_Coding#cite_note-51) > It is not, however, used for high-definition television broadcasts, as the importance of this profile faded when the High Profile was developed in 2004 for that application.> 
>
> __High Profile (HiP, 100)__
>  * The primary profile for broadcast and disc storage applications, particularly for high-definition television applications (for example, this is the profile adopted by the [Blu-ray Disc](https://www.wikipedia.org/wiki/Blu-ray_Disc) storage format and the [DVB](https://www.wikipedia.org/wiki/Digital_Video_Broadcasting) HDTV broadcast service).
> ...

... **EDIT** It looks like you can get the specifications from https://spirespark.com/dlna/guidelines * [citation](https://stackoverflow.com/questions/10502306/where-can-i-find-the-full-dlna-specifications/24272853#24272853)

> 1. ... I think I have the PDFs somewhere

Would be great to know some more basics in case this affects the quality of playback from the device side *(perhaps with more memory allocation for higher quality expected decoding)*.

Thank you too.

I wonder if EU is the equivalent to a region code *(US 1 for DVD and US A for BD perhaps)*. I might go try `US` in there as well with just the default in a bit. **EDIT** No change. Still needs remaps.

**Reply by KarlStraussberger** [reply]
I think the EU goes back to the old broadcasting standards of PAL and NTSC which was 50 or 60 Hz refresh rate and some line count delta, when I was really young. :smile:

**Reply by KarlStraussberger** [reply]
For 2. see #3208
